conductor/src/heartbeat_monitor.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+fn current_timestamp() -> u64 {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("now() is not later than UNIX_EPOCH")
+        .as_secs()
+}
+
+#[derive(Clone)]
+pub struct HeartbeatMonitor {
+    shared_heartbeat: Arc<AtomicU64>,
+    update_interval: Duration,
+}
+
+#[derive(Clone)]
+pub struct HeartbeatUpdater {
+    shared_heartbeat: Arc<AtomicU64>,
```

Comment:
Went with a lock-free approach rather than something like `Arc<RwLock<Instant>>`,  as the issue with workers getting stuck might stem from lock contention, so adding more lock contention probably wouldn't help